addons/mail/static/src/models/messaging_initializer/messaging_initializer.js

```diff
@@ -206,6 +209,45 @@ function factory(dependencies) {
             this.messaging.notificationGroupManager.computeGroups();
         }
 
+        /**
+         * @param {object} resUsersSettings
+         * @param {integer} resUsersSettings.id
+         * @param {boolean} resUsersSettings.is_discuss_sidebar_category_channel_open
+         * @param {boolean} resUsersSettings.is_discuss_sidebar_category_chat_open
+         */
+        _initResUsersSettings({ id, is_discuss_sidebar_category_channel_open, is_discuss_sidebar_category_chat_open }) {
+            this.messaging.update({
+                resUsersSettingsId: id,
+            });
+            this.messaging.discuss.update({
+                categoryChannel: create({
+                    autocompleteMethod: 'channel',
+                    commandAddTitleText: this.env._t("Add or join a channel"),
+                    counterComputeMethod: 'needaction',
+                    hasAddCommand: true,
+                    hasViewCommand: true,
+                    isServerOpen: is_discuss_sidebar_category_channel_open,
+                    name: this.env._t("Channels"),
+                    newItemPlaceholderText: this.env._t("Find or create a channel..."),
+                    serverStateKey: 'is_discuss_sidebar_category_channel_open',
+                    sortComputeMethod: 'name',
+                    supportedChannelTypes: ['channel'],
+                }),
+                categoryChat: create({
+                    autocompleteMethod: 'chat',
+                    commandAddTitleText: this.env._t("Start a conversation"),
+                    counterComputeMethod: 'unread',
+                    hasAddCommand: true,
+                    isServerOpen: is_discuss_sidebar_category_chat_open,
+                    name: this.env._t("Direct Messages"),
+                    newItemPlaceholderText: this.env._t("Find or start a conversation..."),
+                    serverStateKey: 'is_discuss_sidebar_category_chat_open',
+                    sortComputeMethod: 'last_action',
+                    supportedChannelTypes: ['chat'],
+                }),
```

Comment:
Most fields should be computed. There is a few categories, and the field values depend on the type of category, hence most fields should be computed.

Another reason for disliking not doing computed fields, is that this approach of setting all fields is fragile: they are set in this specific flow, can easily be overwritten/erased in other flows, values are too far away from field definition, etc.